micrometer-tracing-tests/micrometer-tracing-test/src/main/java/io/micrometer/tracing/test/simple/SimpleSpan.java

```diff
@@ -177,7 +186,27 @@ public String getRemoteServiceName() {
     }
 
     @Override
-    public FinishedSpan setRemoteServiceName(String remoteServiceName) {
+    public SimpleSpan setRemoteServiceName(String remoteServiceName) {
+        return this;
+    }
+
+    @Override
+    public Map<TraceContext, Map<String, String>> getLinks() {
```

Comment:
Should not we use a `Link` interface everywhere in our public APIs and convert `Link`s to `Map` at the end when we need it?